### PR TITLE
Built-in Sandbox: add more unit tests for dragonball. Part 3

### PR DESCRIPTION
**Pull request by wllenyj**
Added more unit tests for device manager.

Fixes: #4899 

Signed-off-by: wllenyj <wllenyj@linux.alibaba.com>

### CODE REVIEW

File: src/dragonball/src/device_manager/mod.rs [hunk 1]
```diff
@@ -1011,3 +1011,170 @@ impl DeviceManager {
...
+
+    use super::*;
+    use crate::vm::CpuTopology;
+
+    impl DeviceManager {
+        pub fn new_test_mgr() -> Self {
+            let kvm = Kvm::new().unwrap();
+            let vm = kvm.create_vm().unwrap();
+            let vm_fd = Arc::new(vm);
+            //let file = TempFile::new().unwrap().into_file();
```
Comment (liubin) [inline src/dragonball/src/device_manager/mod.rs:1031]: Delete these two lines?

File: src/dragonball/src/device_manager/mod.rs [hunk 2]
```diff
@@ -1011,3 +1011,170 @@ impl DeviceManager {
...
+                logger,
+            }
+        }
+    }
+
+    #[test]
+    fn test_create_device_manager() {
+        skip_if_not_root!();
+        let mgr = DeviceManager::new_test_mgr();
+        let _ = mgr.io_manager();
```
Comment (liubin) [inline src/dragonball/src/device_manager/mod.rs:1066]: This seems will test nothing.

Comment (wllenyj) [reply src/dragonball/src/device_manager/mod.rs:1066]: This is just a simple test that it works.

File: src/dragonball/src/device_manager/mod.rs (hunk 1 above)
Comment (wllenyj) [reply src/dragonball/src/device_manager/mod.rs]: fixed

**Review by liubin — APPROVED** [review]

File: src/dragonball/src/device_manager/mod.rs [hunk 3]
```diff
@@ -1011,3 +1011,168 @@ impl DeviceManager {
...
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_create_legacy_device_manager() {
```
Comment (studychao) [inline src/dragonball/src/device_manager/mod.rs:1069]: is there other redundant unit test called `test_create_legacy_device_manager` under `/device_manager/legacy.rs`?
Should we remove it?

File: src/dragonball/src/device_manager/mod.rs [hunk 4]
```diff
@@ -1011,3 +1011,168 @@ impl DeviceManager {
...
+        skip_if_not_root!();
```
Comment (studychao) [inline src/dragonball/src/device_manager/mod.rs:1070]: why `skip_if_not_root` is here?
I have tried in my dev machine and this test case work even without `sudo`

File: src/dragonball/src/device_manager/mod.rs (hunk 3 above)
Comment (wllenyj) [reply src/dragonball/src/device_manager/mod.rs]: The content of the two tests is different and has been renamed to test_create_devices.
Thanks.

File: src/dragonball/src/device_manager/mod.rs (hunk 4 above)
Comment (wllenyj) [reply src/dragonball/src/device_manager/mod.rs:1070]: The `/device_manager/legacy.rs` 's `test_create_legacy_device_manager` is no privilege required. Renaming the other one fixes it.

**Review by studychao — APPROVED** [review]

**Comment by studychao** [comment]
Thanks:)

**Pull request merged**